File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_4_anti_patterns/after_rich_domain_thin_controller/domain/models/Customer.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_4_anti_patterns.after_rich_domain_thin_controller.domain.models;

/**
 * ARCHITECTURE NOTE: Not every domain model needs complex behavior. 
 * Because the core business rules for this bounded context revolve 
 * around the Order, this Customer class can remain a simple data 
 * entity holding state.
 */
public class Customer {
    private int id;
    private String type; // e.g., "Gold"
    private String email;

    public Customer() {
        // Default constructor
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
}

File: Chapter 04/C#/Section_4_4_Anti_Patterns/After_Rich_Domain_Thin_Controller/After/After.Domain/Models/Customer.cs
namespace After.Domain.Models
{
    /// <summary>
    /// ARCHITECTURE NOTE: Not every domain model needs complex behavior. 
    /// Because the core business rules for this bounded context revolve 
    /// around the Order, this Customer class can remain a simple data 
    /// entity holding state.
    /// </summary>
    public class Customer
    {
        public int Id { get; set; }
        public string Type { get; set; } // e.g., "Gold"
        public string Email { get; set; }
    }
}